### ISSUE TITLE
Invalid values cause crash on MySQL with `withNewConnection`

### ISSUE THREAD

**Issue by t-ae** [vapor/mysql-kit#212]
Moved from vapor repository.
https://github.com/vapor/vapor/issues/1782

If there's invalid value, like 3 for `Bool`, in DB, `MySQLError` will be throws while decoding.
However, if we are using `withNewConenction`, it triggers this `fatalError`.
https://github.com/vapor/mysql/blob/448f1ca28b3ea0f64a627bec22e5b0945d31b9f4/Sources/MySQL/Connection/MySQLConnection.swift#L120

Example:
```swift
struct User: MySQLModel {
    static var entity = "users"
    var id: Int?
    var is_banned: Bool
}

// Suppose there are some rows that have 2 or 3 or something for `is_banned` column.

do {
    // this throws error
    try User.query(on: req).all().wait()
} catch {
    print("use req directly: \(error)")
}

do {
    // this also throws error
    try req.withPooledConnection(to: .mysql) { conn in
        User.query(on: conn).all()
        }
    }.wait()
} catch {
    print("withPooledConnection: \(error)")
}

do {
    // this cause crash
    try req.withNewConnection(to: .mysql) { conn in
        User.query(on: conn).all()
    }.wait()
} catch {
    print("withNewConenction: \(error)")
}
```

**Comment by sofbix** [comment]
For using wait() I've made hack:

```
extension DatabaseConnectable {
    
    func newBackgroundTransaction<T>(_ callback: @escaping (_ connection: DatabaseConnectable) throws -> T) -> EventLoopFuture<T>{
        //return self.transaction(on: .mysql, { (connection) throws  -> EventLoopFuture<T> in
        let connection = self
            let promise = connection.eventLoop.newPromise(T.self)
            DispatchQueue.global().async { () -> Void in
                do {
                    promise.succeed(result: try callback(connection))
                } catch (let error) {
                    promise.fail(error: error)
                }
            }
            return promise.futureResult
        //})
    }
}
```

**Comment by tanner0101** [comment]
This has been fixed in MySQLNIO 1.0.0.
